lib/rake/test.rake

```diff
@@ -0,0 +1,99 @@
+# Run 'rake' or 'rake -P' to get a list of valid Rake commands.
+
+require 'cdo/hip_chat'
+require 'cdo/test_run_utils'
+require 'cdo/rake_utils'
+require 'cdo/git_utils'
+
+namespace :test do
+  desc 'Runs apps tests.'
+  task :apps do
+    TestRunUtils.run_apps_tests
+  end
+
+  desc 'Runs code studio tests.'
+  task :code_studio do
+    TestRunUtils.run_code_studio_tests
+  end
+
+  desc 'Runs blockly-core tests.'
+  task :blockly_core do
+    TestRunUtils.run_blockly_core_tests
+  end
+
+  desc 'Runs dashboard tests.'
+  task :dashboard do
+    TestRunUtils.run_dashboard_tests
+  end
+
+  desc 'Runs pegasus tests.'
+  task :pegasus do
+    TestRunUtils.run_pegasus_tests
+  end
+
+  desc 'Runs shared tests.'
+  task :shared do
+    TestRunUtils.run_shared_tests
+  end
+
+  namespace :changed do
+    desc 'Runs apps tests if apps might have changed from staging.'
+    task :apps do
+      run_tests_if_changed('apps', ['apps/*', 'blockly-core/*', 'shared/*.js', 'shared/*.css']) do
```

Comment:
Oh, interesting, via [The Ruby Programming Language](https://books.google.com/books?id=jcUbTcr5XWwC&pg=PA352&lpg=PA352&dq=ruby+file+fnmatch&source=bl&ots=fJGjxi9wfz&sig=x756POI9GLtlIDmmb2cYOyJsm7g&hl=en&sa=X&ved=0ahUKEwjz0-6kmufMAhUpw4MKHVXjCkQQ6AEIPjAE#v=onepage&q=ruby%20file%20fnmatch&f=false):

> fnmatch should usually be invoked with a third argument of `File::FNM_PATHNAME`

Maybe it would make more sense to include some of those options and be explicit about matching ./ and not globbing slashes to more closely match expected behavior?